### ISSUE TITLE
Support EWS or EAS

### ISSUE THREAD

**Issue by mathieumd** [thundernest/k-9#6119]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same feature request.
- [X] I will describe the problem with as much detail as possible.
- [X] This issue only contains a request for one single feature, **not** multiple (related) features.

### App version

6.000

### Problem you are trying to solve

My current work uses Zimbra as mail backend, but the smartphone used by employees unfortunately have to use Gmail app because of ActiveSync (Zimbra [EAS](https://www.zimbra.com/business-email-collaboration/secure-email-features/email-exchange-activesync/) or [EWS](https://www.zimbra.com/business-email-collaboration/secure-email-features/email-exchange-web-services/)).

I saw https://github.com/thundernest/k-9/issues/606#issuecomment-526831943, but now that [K-9 Mail joins the Thunderbird family](https://k9mail.app/2022/06/13/K-9-Mail-and-Thunderbird.html) (congrats, by the way! :tada:), maybe EWS/EAS could be reconsidered?

### Suggested solution

Supporting EWS and/or EAS.

**Comment by cketti** [comment]
This is unlikely to happen. Implementing a proprietary protocol will always be at the bottom of the TODO list. And that list is very long.

**Comment by mathieumd** [comment]
I understand. And I'm not going to praise proprietary software! But, on the other side, corporate adoption of K9M is lower than it could due to this missing feature. At least at my place it's clearly the case. Maybe joining Thunderbird could be the opportunity to launch something like "sponsored features"? I know my employer is very likely to pay (maybe not alone, though) for missing features it wants.

**Comment by rezazarchi** [comment]
They have an old branch with initial EAS support: https://github.com/k9mail/k-9/tree/ms-eas
It might be helpful for you.

**Comment by Arndorferd** [comment]
I agree that this should be priority. Not everyone chooses what emails they get at times, I have a school email that requires ews and there are no android mail clients that have the feature but evolution works on desktop.

**Comment by ASLLR** [comment]
Hello, i'm intereted too by this funtionnality. The link of old branch return a 404 error. 

Best regards

**Comment by tdboorman** [comment]
I personally think that this should be higher up the to do list. I do not want to have to use two email apps on my phone, which I currently have to do as IMAP access has been removed for one of my Office 365 accounts.

**Comment by hex-m** [comment]
Thunderbird (for Desktop) [saying](https://blog.thunderbird.net/2024/05/thunderbird-monthly-development-digest-may-2024/)

> We’re very very very close!

Makes this a brand issue. If both clients *are Thunderbird* we should strive to support the same feature-set to avoid user confusion and enable a seamless cross-platform UX.

**Comment by Sanmilie** [comment]
As a dedicated user and ardent supporter of your application, I want to express my gratitude for the exceptional work you do. K-9 Mail is a powerful and flexible email client that meets the needs of many Android users. However, I would like to suggest an enhancement that could significantly benefit a large portion of your user base: support for Exchange ActiveSync (EAS) and /or Exchange Web Services (EWS) protocols.

Background and Justification
In the professional world, many companies use Exchange servers to manage their electronic communications. These environments often prioritize the use of EAS and EWS protocols for synchronizing emails, calendars, contacts, and tasks. Unfortunately, many of these companies impose strict restrictions on using the SMTP / IMAP protocol for incoming and sending emails, limiting the capabilities of some email clients, including K-9 Mail.

Benefits of EAS/EWS Support for K-9 Mail
Increased Compatibility with Enterprise Environments:

EAS/EWS support would enable K-9 Mail to integrate more seamlessly with Exchange servers, meeting the needs of professional users whose companies restrict email sending via SMTP and the IMAP protocol.
Comprehensive Synchronization:

EAS and EWS handle not only emails but also offer synchronization for calendars, contacts, and tasks, enhancing the overall user experience and productivity.
Enhanced Security:

EAS and EWS protocols are designed with robust security mechanisms, meeting the stringent requirements of enterprise IT environments.
Attractiveness to New Users:

By adding support for EAS/EWS, K-9 Mail could attract a broader audience, particularly among professionals and businesses looking for a reliable and comprehensive email solution for Android.
Conclusion
Integrating support for EAS and EWS protocols into K-9 Mail would represent a significant advancement for the application. It would not only improve its compatibility with enterprise environments but also offer a more comprehensive and secure email solution to users. As a dedicated user of K-9 Mail, I strongly encourage you to consider this request and explore the possibilities of implementing these protocols.

**Comment by Konfekt** [comment]
[Aquamail](https://aquamail.freshdesk.com/support/home) provides this. Are there other alternatives? For example, is there on Android something akin to [Davmail](https://github.com/mguessan/davmail/) that provides a mail server for Exchange Web Services (EWS) protocols that mail clients can connect to?

**Comment by kewisch** [comment]
Given the wealth of users who are forced to use this there is a slightly higher chance we might get to this. We'll let folks know here when we get to it. If you want to discuss alternatives that already support this you are welcome to, though the bug tracker is likely not the best place for it.

**Comment by KGS501** [comment]
Now that the desktop app finally supports Exchange accounts natively, is there a timeline when this feature will be available in the android app?
I recently tried Thunderbird for the first time and am very pleased (with the desktop app) but that's the last thing keeping me from making Thunderbird my main mail service and replacing Gmail.

**Comment by slu-010101** [comment]
Native ActiveSync support in Thunderbird Android would be very nice.
From admin view, ActiveSync for mobiles is still the best option, only one port for everything (mail,address,calendar).

**Comment by parislamb** [comment]
I would love to see exchange support on mobile. My university email uses sharepoint/microsoft and support for open protocols has been deprecated afaik so i'm stuck using multiple email clients. I know the focus is on re-architecture for now and that further development will be easier once that has been completed but I hope this gets implemented sooner than later. Really appreciate all the hard work the team puts in to this project. Some of the best and most consistent open source software out there.
